src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,47 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
+
+  // died() {
+  //   if (this.health <= 0) {
+  //     delete Animal.alive.this;
+  //   }
+  // }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.name = name;
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+
+    this.name = name;
+  }
+
+  bite(name) {
+    if (name.hidden === false) {
+      name.health -= 50;
```

Comment:
After decreasing the health of the herbivore, there should be a check to see if its health is less than or equal to 0. If so, the herbivore should be removed from the `Animal.alive` array at this point, rather than filtering the array every time the `bite` method is called.